adds frank to the homepage

frankV opened the pull request.

svvitale commented:
Actually, @iandouglas, the HAML *does* get auto-compiled during deployment.

svvitale commented:
@frankV you'll want to update the README.md too.

iandouglas commented:
in that case, carry on :) Travis-CI is failing tho.

svvitale commented:
https://docs.travis-ci.com/user/environment-variables#Defining-encrypted-variables-in-.travis.yml

> Encrypted environment variables are not available to pull requests from forks due to the security risk of exposing such information to unknown code.

Need to figure out the best way to store deployment credentials.  Maybe I move them into travis and out of the .travis.yml file.  You guys have any experience with this? (I'm a Bitbucket Pipelines/CodeShip user)

frankV commented:
Travis has an encryption method so you can put the keys right in your
.travis file. It's like their low tech secrets mgmt

On Fri, Aug 11, 2017 at 10:09 AM Scott Vitale <notifications@github.com>
wrote:

>
> https://docs.travis-ci.com/user/environment-variables#Defining-encrypted-variables-in-.travis.yml
>
> Encrypted environment variables are not available to pull requests from
> forks due to the security risk of exposing such information to unknown code.
>
> Need to figure out the best way to store deployment credentials. Maybe I
> move them into travis and out of the .travis.yml file. You guys have any
> experience with this? (I'm a Bitbucket Pipelines/CodeShip user)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/boulder-python/boulderpython.org/pull/1#issuecomment-321854256>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAev9JfAGfdpA8zdIAxraa-7x0TcruOMks5sXHy1gaJpZM4O0zwB>
> .
>
-- 
Frank Valcarcel
Cuttlesoft <http://www.cuttlesoft.com>
888.534.1844
<http://www.cuttlesoft.com>

frankV commented:
🤔 I think it only works with services they integrate.

On Fri, Aug 11, 2017 at 10:11 AM Frank Valcarcel <frank@cuttlesoft.com>
wrote:

> Travis has an encryption method so you can put the keys right in your
> .travis file. It's like their low tech secrets mgmt
>
> On Fri, Aug 11, 2017 at 10:09 AM Scott Vitale <notifications@github.com>
> wrote:
>
>>
>> https://docs.travis-ci.com/user/environment-variables#Defining-encrypted-variables-in-.travis.yml
>>
>> Encrypted environment variables are not available to pull requests from
>> forks due to the security risk of exposing such information to unknown code.
>>
>> Need to figure out the best way to store deployment credentials. Maybe I
>> move them into travis and out of the .travis.yml file. You guys have any
>> experience with this? (I'm a Bitbucket Pipelines/CodeShip user)
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/boulder-python/boulderpython.org/pull/1#issuecomment-321854256>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AAev9JfAGfdpA8zdIAxraa-7x0TcruOMks5sXHy1gaJpZM4O0zwB>
>> .
>>
> --
> Frank Valcarcel
> Cuttlesoft <http://www.cuttlesoft.com>
> 888.534.1844
> <http://www.cuttlesoft.com>
>
-- 
Frank Valcarcel
Cuttlesoft <http://www.cuttlesoft.com>
888.534.1844
<http://www.cuttlesoft.com>

svvitale commented:
Oh!  We won't be deploying from pull requests anyway.  The credentials should only be used in the deploy section.

Can you try moving these two lines to the deploy section?
```
# Decrypt the credentials we added to the repo using the key we added with the Travis command line tool
- openssl aes-256-cbc -K $encrypted_3a75846c8fd3_key -iv $encrypted_3a75846c8fd3_iv -in client-secret.json.enc -out client-secret.json -d
# Here we use the decrypted service account credentials to authenticate the command line tool
- gcloud auth activate-service-account --key-file client-secret.json
```

If it gets more complicated, I can try and work out the build issues apart from this PR.

frankV commented:
should it look like this?

```
deploy:
  # Deploy the app
  skip_cleanup: true
  provider: script
  script: 
    # Decrypt the credentials we added to the repo using the key we added with the Travis command line tool
    - openssl aes-256-cbc -K $encrypted_3a75846c8fd3_key -iv $encrypted_3a75846c8fd3_iv -in client-secret.json.enc -out client-secret.json -d
    # Here we use the decrypted service account credentials to authenticate the command line tool
    - gcloud auth activate-service-account --key-file client-secret.json
    - gcloud -q app deploy app.yaml --promote --verbosity=info```

svvitale commented:
Yes, except for those three trailing back quotes :)

frankV commented:
Do we need to install openssl to travis' environment?

frankV commented:
According to this: https://github.com/pockethub/PocketHub/issues/884

Those environment variables `$encrypted_3a75846c8fd3_key` and `$encrypted_3a75846c8fd3_iv` aren't available in pull-requests or forks, so we may need to complete the merge in order for it to deploy.

svvitale commented:
You'll also want to remove those two lines from the `before_install` section.

svvitale commented:
Alright, this is officially more of a pain in the ass.  I'm okay with merging and fixing Travis separately.
